src/utils/variable.ts

```diff
@@ -60,7 +60,9 @@ export const selectVariableValues = (values: string[], runtimeVariable?: Runtime
          */
         if (searchParams.length === 0 && !locationService.getSearchObject()[`var-${name}`]) {
           searchParams.push(
-            ...runtimeVariable.options.filter((option) => option.selected).map((option) => option.text)
+            ...runtimeVariable.options
+              .filter((option) => option.selected)
+              .map((option) => (option.value === AllValueParameter ? option.text : option.value))
```

Comment:
Reason why new options had appeared, we used option.text instead of value. Text should be used only for all option

Reply:
Thank you.